frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watcher.ts

```diff
@@ -0,0 +1,79 @@
+import * as _ from 'lodash';
+// import * as k8sActions from '@console/internal/actions/k8s';
+import { K8sModel } from '../../../api/common-types';
+import * as k8sActions from '../../../app/k8s/actions/k8s';
+import { SDKStoreState } from '../../../app/redux-types';
+import { WatchK8sResource } from '../../../extensions/console-types';
+import { getReferenceForModel } from '../k8s-ref';
+import { GetIDAndDispatch, MakeQuery, Query } from './k8s-watch-types';
+
+export class NoModelError extends Error {
```

Comment:
FYI - in OpenShift plugin SDK we already have a [`CustomError`](https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-runtime/src/utils/errors.ts) which is a port of OpenShift Console [`CustomError`](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/utils/error/custom-error.ts) but without
```ts
Object.setPrototypeOf(this, new.target.prototype);
```
since that isn't needed for ES6+ code target.

Custom error classes _should_ extend from `CustomError` instead of base `Error` though, can we do this change here?

Reply:
Agree with @vojtechszocs good catch.

Reply:
Done!